edx_proctoring/api.py

```diff
@@ -0,0 +1,558 @@
+    },
+    {
+        ...: ...,
+        ...: ...
+
+    },
+    ..
+    ]
+    """
+    exams = ProctoredExam.get_all_exams_for_course(course_id)
+
+    return [ProctoredExamSerializer(proctored_exam).data for proctored_exam in exams]
+
+
+def get_all_exam_attempts(course_id):
+    """
+    Returns all the exam attempts for the course id.
+    """
+    exam_attempts = ProctoredExamStudentAttempt.objects.get_all_exam_attempts(course_id)
+    return [ProctoredExamStudentAttemptSerializer(active_exam).data for active_exam in exam_attempts]
+
+
+def get_filtered_exam_attempts(course_id, search_by):
+    """
+    returns all exam attempts for a course id filtered by  the search_by string in user names and emails.
+    """
+    exam_attempts = ProctoredExamStudentAttempt.objects.get_filtered_exam_attempts(course_id, search_by)
+    return [ProctoredExamStudentAttemptSerializer(active_exam).data for active_exam in exam_attempts]
+
+
+def get_active_exams_for_user(user_id, course_id=None):
+    """
+    This method will return a list of active exams for the user,
+    i.e. started_at != None and completed_at == None. Theoretically there
+    could be more than one, but in practice it will be one active exam.
+
+    If course_id is set, then attempts only for an exam in that course_id
+    should be returned.
+
+    The return set should be a list of dictionary objects which are nested
+
+
+    [{
+        'exam': <exam fields as dict>,
+        'attempt': <student attempt fields as dict>,
+        'allowances': <student allowances as dict of key/value pairs
+    }, {}, ...]
+
+    """
+    result = []
+
+    student_active_exams = ProctoredExamStudentAttempt.objects.get_active_student_attempts(user_id, course_id)
+    for active_exam in student_active_exams:
+        # convert the django orm objects
+        # into the serialized form.
+        exam_serialized_data = ProctoredExamSerializer(active_exam.proctored_exam).data
+        active_exam_serialized_data = ProctoredExamStudentAttemptSerializer(active_exam).data
+        student_allowances = ProctoredExamStudentAllowance.get_allowances_for_user(
+            active_exam.proctored_exam.id, user_id
+        )
+        allowance_serialized_data = [ProctoredExamStudentAllowanceSerializer(allowance).data for allowance in
+                                     student_allowances]
+        result.append({
+            'exam': exam_serialized_data,
+            'attempt': active_exam_serialized_data,
+            'allowances': allowance_serialized_data
+        })
+
+    return result
+
+
+def get_student_view(user_id, course_id, content_id, context):  # pylint: disable=too-many-branches
+    """
+    Helper method that will return the view HTML related to the exam control
+    flow (i.e. entering, expired, completed, etc.) If there is no specific
+    content to display, then None will be returned and the caller should
+    render it's own view
+    """
+
+    has_started_exam = False
```

Comment:
It seems this declaration will never actually get used, because the value is being overwritten on line 502 prior to any references -- so it seems unnecessary -- have you run coverage.py and checked the branch coverage?  I'd think this one would come up as a branch miss (the solution being to simply remove it).